### PR TITLE
Support More Aggregate Methods

### PR DESCRIPTION
**Pull request by elvaliuliuliu**
This PR exposes more aggregate methods for `RelationalGroupedDataset` like `Mean()`, `Max()`, `Avg()`, `Min()`. Follow up on  issue #260

### CODE REVIEW

File: src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs [hunk 1]
```diff
@@ -44,6 +44,34 @@ internal RelationalGroupedDataset(JvmObjectReference jvmObject, DataFrame dataFr
         public DataFrame Count() =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("count"));
 
+        /// <summary>
+        /// Compute the mean value for each numeric columns for each group.
+        /// </summary>
+        /// <returns>New DataFrame object with mean applied</returns>
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs]: Let's add <param name="colNames">..</param> to each of these.  Can you also add it to Sum() as well.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs [hunk 2]
```diff
@@ -404,6 +404,22 @@ public void TestSignaturesV2_3_X()
             Assert.IsType<RelationalGroupedDataset>(_df.GroupBy(_df["age"]));
             Assert.IsType<RelationalGroupedDataset>(_df.GroupBy(_df["age"], _df["name"]));
 
+            Assert.IsType<DataFrame>(_df.GroupBy("name").Mean("age"));
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: @imback82  should we have a separate test suite for RelationalGroupedDataset or are these fine here ?

File: src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs (hunk 1 above)
Comment (elvaliuliuliu) [reply src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs]: Added

File: src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs [hunk 3]
```diff
@@ -44,9 +44,42 @@ internal RelationalGroupedDataset(JvmObjectReference jvmObject, DataFrame dataFr
         public DataFrame Count() =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("count"));
 
+        /// <summary>
+        /// Compute the mean value for each numeric columns for each group.
+        /// </summary>
+        /// <param name="colNames">Column to compute mean on</param>
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs]: Name of columns to ...

Comment (elvaliuliuliu) [reply src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs]: Gotcha!

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs (hunk 2 above)
Comment (imback82) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: fine for now.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs [hunk 4]
```diff
@@ -404,6 +404,22 @@ public void TestSignaturesV2_3_X()
             Assert.IsType<RelationalGroupedDataset>(_df.GroupBy(_df["age"]));
             Assert.IsType<RelationalGroupedDataset>(_df.GroupBy(_df["age"], _df["name"]));
 
+            Assert.IsType<DataFrame>(_df.GroupBy("name").Mean("age"));
+            Assert.IsType<DataFrame>(
+                _df.WithColumn("tempAge", _df["age"]).GroupBy("name").Mean("age", "tempAge"));
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: for these GroupBy.fn tests, let's pull out _df.WithColumn("tempAge", _df["age"]).GroupBy("name") into a var within it's own scope, { ... } and reuse.

Comment (suhsteve) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: Same with _df.GroupBy("name") as well.

Comment (elvaliuliuliu) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: I have made the changes. Please let me know if the var name makes sense. Thanks!

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs [hunk 5]
```diff
@@ -404,9 +404,23 @@ public void TestSignaturesV2_3_X()
             Assert.IsType<RelationalGroupedDataset>(_df.GroupBy(_df["age"]));
             Assert.IsType<RelationalGroupedDataset>(_df.GroupBy(_df["age"], _df["name"]));
 
-            Assert.IsType<DataFrame>(_df.GroupBy("name").Sum("age"));
-            Assert.IsType<DataFrame>(
-                _df.WithColumn("tempAge", _df["age"]).GroupBy("name").Sum("age", "tempAge"));
+            var relationalGroupedDataset = _df.GroupBy("name");
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: Let's enclose this whole subsection with a { }.  We can shorten the var names.  Just using `df1` and `df2` should suffice.

Comment (elvaliuliuliu) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs]: Gotcha! Done

**Pull request merged**